src/adapters/MorphoMarketV1Adapter.sol

```diff
@@ -119,15 +153,21 @@ contract MorphoMarketV1Adapter is IMorphoMarketV1Adapter {
                     break;
                 }
             }
-        } else if (oldAllocation == 0 && newAllocation > 0) {
+        } else if (_oldAllocation == 0 && _newAllocation > 0) {
             marketParamsList.push(marketParams);
         }
     }
 
-    function allocation(MarketParams memory marketParams) public view returns (uint256) {
+    function oldAllocation(MarketParams memory marketParams) public view returns (uint256) {
         return IVaultV2(parentVault).allocation(keccak256(abi.encode("this/marketParams", address(this), marketParams)));
     }
 
+    function newAllocation(MarketParams memory marketParams) public view returns (uint256) {
+        (uint256 totalSupplyAssets, uint256 totalSupplyShares,,) =
```

Comment:
We could save 4200 gas (2 cold sloads) per market by directly reading a slot instead of calling market(id).

Reply:
Nevermind it wouldn't be up to date since you need to take the interests into account.